elf/compat.go

```diff
@@ -0,0 +1,70 @@
+// +build linux
+
+// (c) 2018 Suchakra Sharma <suchakra@shiftleft.io>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elf
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"runtime"
+)
+
+func GetSyscallFnName(name string) (string, error) {
+	var arch string
+	if runtime.GOARCH == "amd64" {
```

Comment:
you can implement this if chain with a switch:

```go
switch runtime.GOARCH {
  case "386": arch = "ia32"
  default: arch = "x64"
}
```

It's more compact and clear

Reply:
Thanks, will do this :+1: